File: Android-App/app/src/main/java/com/ite393group5/android_app/models/LocationInfo.kt
package com.ite393group5.android_app.models

import kotlinx.serialization.Serializable

@Serializable
data class LocationInfo(
    val houseNumber:String,
    val street:String,
    val zone:String,
    val barangay:String,
    val cityMunicipality:String,
    val province:String,
    val country:String,
    val postalCode:String,
)

File: web/includes/models/LocationInfo.php
<?php
namespace models;
class LocationInfo {
    public ?string $houseNumber;
    public ?string $street;
    public ?string $zone;
    public ?string $barangay;
    public ?string $cityMunicipality;
    public ?string $province;
    public ?string $country;
    public ?string $postalCode;

    public function __construct($data) {
        $this->houseNumber = $data['houseNumber'] ?? null;
        $this->street = $data['street'] ?? null;
        $this->zone = $data['zone'] ?? null;
        $this->barangay = $data['barangay'] ?? null;
        $this->cityMunicipality = $data['cityMunicipality'] ?? null;
        $this->province = $data['province'] ?? null;
        $this->country = $data['country'] ?? null;
        $this->postalCode = $data['postalCode'] ?? null;
    }

    public function toArray() {
        return get_object_vars($this);
    }
}
?>
